modules/flann/include/opencv2/flann/heap.h

```diff
@@ -36,6 +36,11 @@
 #include <algorithm>
 #include <vector>
 
+#include <unordered_map>
+#include <mutex>
```

Comment:
> mutex

There is special build mode of OpenCV for single-threaded binaries:
- #19985

So please avoid manual using of mutex/thread headers and use `cv::Mutex` / `cv::AutoLock` OpenCV wrappers.